### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by omjadhav85**
Features added:

1. User can now create account.
2. User can now log in.
3. Added Authentication context for efficient use of frequently used variables.
4. Added Redirect and PrivateRoute router components for auth checking and to navigate to proper routes as and when required.

### CODE REVIEW

**Review by ajay-mandaviya — APPROVED** [review]
Everything looks good no changes require 
good to merge

File: src/pages/Authentication/Login.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  //   const { dispatchData } = useData();
+  const navigate = useNavigate();
+  const location = useLocation();
+  const gotoPath = location.state?.from?.pathname || "/";
+
+  const testData = {
+    email: "adarshbalika@gmail.com",
+    password: "adarshBalika123",
+  };
+
```
Comment (junaidshaikh-js) [inline src/pages/Authentication/Login.js]: You can change the default user to your own name and email

**Review by junaidshaikh-js — APPROVED** [review]
Good work. I liked how you made separate component for input and generic div 

Suggested one change, everything else looks fine.

File: src/pages/Authentication/Login.js (hunk 1 above)
Comment (omjadhav85) [reply src/pages/Authentication/Login.js]: Thanks for this suggestion Junaid! made the changes

**Pull request merged**